src/main/java/de/zalando/aruha/nakadi/controller/SubscriptionStreamController.java

```diff
@@ -0,0 +1,125 @@
+                    } else {
+                        writeProblemResponse(Response.Status.SERVICE_UNAVAILABLE, "Failed to continue streaming");
+                    }
+                } catch (final IOException e) {
+                    LOG.error("Failed to write exception to response", e);
+                }
+            } else {
+                LOG.warn("Exception found while streaming, but no data could be provided to client", ex);
+            }
+        }
+
+        void writeProblemResponse(final Response.StatusType statusCode, final String message) throws IOException {
+            response.setStatus(statusCode.getStatusCode());
+            response.setContentType("application/problem+json");
+            jsonMapper.writer().writeValue(out, Problem.valueOf(statusCode, message));
+        }
+
+        @Override
+        public void streamData(final byte[] data) throws IOException {
+            headersSent = true;
+            out.write(data);
+            out.flush();
+        }
+    }
+
+    @RequestMapping(value = "/subscriptions/{subscription_id}/events", method = RequestMethod.GET)
+    public StreamingResponseBody streamEvents(
+            @PathVariable("subscription_id") final String subscriptionId,
+            @RequestParam(value = "window_size", required = false, defaultValue = "100") final int windowSize,
+            @RequestParam(value = "commit_timeout", required = false, defaultValue = "30") final int commitTimeout,
+            @RequestParam(value = "batch_limit", required = false, defaultValue = "1") final int batchLimit,
+            @RequestParam(value = "stream_limit", required = false) final Integer streamLimit,
+            @RequestParam(value = "batch_flush_timeout", required = false, defaultValue = "30") final int batchTimeout,
+            @RequestParam(value = "stream_timeout", required = false) final Long streamTimeout,
+            @RequestParam(value = "stream_keep_alive_limit", required = false) final Integer streamKeepAliveLimit,
+            final NativeWebRequest request, final HttpServletResponse response) throws IOException {
+
+        return outputStream -> {
+            SubscriptionStreamer streamer = null;
+            final SubscriptionOutputImpl output = new SubscriptionOutputImpl(response, outputStream);
+            try {
+                streamer = subscriptionStreamerFactory.build(
+                        subscriptionId,
+                        StreamParameters.of(batchLimit, streamLimit, batchTimeout, streamTimeout, streamKeepAliveLimit, windowSize, commitTimeout),
+                        output);
+                streamer.stream();
+            } catch (final NoSuchSubscriptionException e) {
+                output.onException(e);
+            } catch (final InterruptedException ex) {
+                LOG.warn("Interrupted while streaming with " + streamer, ex);
+                Thread.currentThread().interrupt();
```

Comment:
Because catching of InterruptedException clears Thread's interrupted flag. If I won't do that, the guy that is stopping this thread will fail.